examples/asr_training/rnnt_lightning.py

```diff
@@ -0,0 +1,535 @@
+                CustomDataset(
+                    torchaudio.datasets.LIBRISPEECH(
+                        self.librispeech_path, url="dev-other"
+                    ),
+                    1000,
+                ),
+            ]
+        )
+        dataloader = torch.utils.data.DataLoader(
+            dataset, batch_size=None, collate_fn=self._valid_collate_fn, num_workers=10,
+        )
+        return dataloader
+
+    def test_dataloader(self):
+        dataset = torchaudio.datasets.LIBRISPEECH(
+            self.librispeech_path, url="test-clean"
+        )
+        dataloader = torch.utils.data.DataLoader(
+            dataset, batch_size=1, collate_fn=self._test_collate_fn
+        )
+        return dataloader
+
+
+def compute_word_level_distance(seq1, seq2):
+    return torchaudio.functional.edit_distance(
+        seq1.lower().split(), seq2.lower().split()
+    )
+
+
+def post_process_hypos(
+    hypos: List[Hypothesis], sp_model: spm.SentencePieceProcessor, tgt_dict: Dictionary
+) -> List[Tuple[str, float, List[int], List[int]]]:
+    post_process_remove_list = [
+        sp_model.unk_id(),
+        sp_model.eos_id(),
+        sp_model.pad_id(),
+    ]
+    hypos_str = [
+        tgt_dict.string(
+            [
+                token_index
+                for token_index in h.tokens[1:]
+                if token_index not in post_process_remove_list
+            ]
+        )
+        for h in hypos
+    ]
+    hypos_str = [sp_model.DecodePieces(s.split()) for s in hypos_str]
+    hypos_ali = [h.alignment[1:] for h in hypos]
+    hypos_ids = [h.tokens[1:] for h in hypos]
+    hypos_score = [[math.exp(h.score)] for h in hypos]
+
+    nbest_batch = list(zip(hypos_str, hypos_score, hypos_ali, hypos_ids))
+
+    return nbest_batch
+
+
+def run_eval(args):
+    model = (
+        RNNTModule.load_from_checkpoint(
+            args.checkpoint_path,
+            librispeech_path=args.librispeech_path,
+            sp_model_path=args.sp_model_path,
+            tgt_dict_path=args.tgt_dict_path,
+            global_stats_path=args.global_stats_path,
+        )
+        .eval()
+        .to(device="cuda")
+    )
+    total_edit_distance = 0
+    total_length = 0
+    dataloader = model.test_dataloader()
+    with torch.no_grad():
+        for idx, (batch, sample) in enumerate(dataloader):
+            actual = sample[0][2]
+            predicted = model(batch)
+            total_edit_distance += compute_word_level_distance(actual, predicted)
+            total_length += len(actual.split())
+            if idx % 100 == 0:
+                logger.info(
+                    f"Processed elem {idx}; WER: {total_edit_distance / total_length}"
+                )
+    logger.info(f"Final WER: {total_edit_distance / total_length}")
+
+
+def run_train(args):
+    checkpoint_dir = args.exp_dir / "checkpoints"
+    checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="Losses/val_loss",
+        mode="min",
+        save_top_k=10,
```

Comment:
10 might be a lot. Or is it necessary to get the best model from the 10 checkpoints?

Reply:
probably not necessary